src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -557,26 +558,36 @@ private boolean validateTemporaryAccessTokenForCourse(String temporaryAccessToke
      */
     @NotNull
     private ResponseEntity<String> handleSaveFile(MultipartFile file, boolean keepFileName, boolean markdown) throws URISyntaxException {
-        // NOTE: Maximum file size is set in resources/config/application.yml
-        // Currently set to 10 MB
+        // NOTE: Maximum file size is set in resources/config/application.yml (currently set to 10 MB)
 
-        // check for file type
         String filename = file.getOriginalFilename();
         if (filename == null) {
-            throw new IllegalArgumentException("Filename cannot be null");
+            throw new IllegalArgumentException("File name can not be null!");
         }
-        // sanitize the filename and replace all invalid characters with "_"
+
+        // Sanitize the filename and replace all invalid characters with an underscore
         filename = filename.replaceAll("[^a-zA-Z\\d\\.\\-]", "_");
-        String fileExtension = FilenameUtils.getExtension(filename);
-        if (this.allowedFileExtensions.stream().noneMatch(fileExtension::equalsIgnoreCase)) {
-            return ResponseEntity.badRequest().body("Unsupported file type! Allowed file types: " + String.join(", ", this.allowedFileExtensions));
+
+        // Check the allowed file extensions
+        final String fileExtension = FilenameUtils.getExtension(filename);
+        final Set<String> allowedExtensions;
+
+        if (markdown) {
+            allowedExtensions = allowedMarkdownFileExtensions;
+        }
+        else {
+            allowedExtensions = allowedFileExtensions;
+        }
+
+        if (allowedExtensions.stream().noneMatch(fileExtension::equalsIgnoreCase)) {
```

Comment:
```suggestion
        if (!allowedExtensions.contains(fileExtension.toLowerCase())) {
```
This seams more intuitive to me

Reply:
Agreed 👍